test/unittests/test_dd_synthesis.cpp

```diff
@@ -0,0 +1,72 @@
+#include "algorithms/synthesis/dd_synthesis.hpp"
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestDDSynth: public testing::TestWithParam<std::string> {
+protected:
+    TruthTable                     tt{};
+    std::string                    testCircuitsDir = "./circuits/";
+    std::unique_ptr<dd::Package<>> ddSynth         = std::make_unique<dd::Package<>>(9U);
+    std::string                    fileName;
+
+    void SetUp() override {
+        fileName = testCircuitsDir + GetParam() + ".pla";
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(TestDDSynth, TestDDSynth,
+                         testing::Values(
+                                 "cnot",
+                                 "swap",
+                                 "toffoli",
+                                 "x2Bit",
+                                 "test_dd_synthesis_1",
+                                 "test_dd_synthesis_2",
+                                 "3_17_6",
+                                 "bitwiseXor2Bit",
+                                 "adder2Bit",
+                                 "adder3Bit",
+                                 "4_49_7",
+                                 "hwb4_12",
+                                 "hwb5_13",
+                                 "hwb6_14",
+                                 "hwb7_15",
+                                 "hwb8_64",
+                                 "hwb9_65"),
+                         [](const testing::TestParamInfo<TestDDSynth::ParamType>& info) {
+                             auto s = info.param;
+                             std::replace( s.begin(), s.end(), '-', '_');
+                             return s; });
+
+TEST_P(TestDDSynth, GenericDDSynthesisTest) {
+    EXPECT_TRUE(readPla(tt, fileName));
+
+    extend(tt);
+
+    auto ttDD = buildDD(tt, ddSynth);
+
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    EXPECT_NE(ttDD, ddSynth->makeIdent(tt.nInputs()));
+
+    DDSynthesizer synthesizer(tt.nInputs());
+
+    const auto srcCpy = ttDD;
```

Comment:
This copy should not be necessary. The synthesis function is not supposed to change the DD edge it receives as input. Furthermore, the check that the DD of the resulting circuit matches the original DD already implies that the synthesis was able to transform the initial DD to the identity. So the last check in this test is redundant.

Reply:
done 👍🏻